### PR TITLE
[INLONG-778]Fix Go SDK Consumer Bug

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes [Jira Issue](https://issues.apache.org/jira/projects/INLONG/issues): https://issues.apache.org/jira/browse/INLONG-778

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by TszKitLo40** [comment]
@gosonzhang @charlely PTAL

**Review by gosonzhang — APPROVED** [review]
+1

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 1]
```diff
@@ -120,7 +120,8 @@ func (c *consumer) register2Master(needChange bool) error {
 		}
 		c.master = node
 	}
-	for c.master.HasNext {
+	retryCount := 0
+	for {
 		rsp, err := c.sendRegRequest2Master()
 		if err != nil {
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:126]: Why here if get error will return err, but not use next master address to register?

Comment (charlely) [reply tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:126]: If first address net error, will always return error ?

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 2]
```diff
@@ -120,7 +120,8 @@ func (c *consumer) register2Master(needChange bool) error {
...
 			log.Infof("[CONSUMER]register2Master error %s", err.Error())
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:127]: Here need use Error level log.

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 3]
```diff
@@ -143,7 +148,7 @@ func (c *consumer) register2Master(needChange bool) error {
 
 		c.masterHBRetry = 0
 		c.processRegisterResponseM2C(rsp)
-		return nil
+		break
 	}
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:152]: If last fail to register to master, need printf fail result to Error log.

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 4]
```diff
@@ -223,17 +228,18 @@ func (c *consumer) GetMessage() (*ConsumerResult, error) {
 	rsp, err := c.client.GetMessageRequestC2B(ctx, m, c.subInfo, c.rmtDataCache)
 	if err != nil {
 		log.Infof("[CONSUMER]GetMessage error %s", err.Error())
+		err1 := c.rmtDataCache.ReleasePartition(true, isFiltered, confirmContext, false)
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:231]: This way can use like it:
if err := c.rmtDataCache.ReleasePartition(true, isFiltered, confirmContext, false); err != nil {
}

Comment (charlely) [reply tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:231]: Why to ReleasePartition?

File: tubemq-client-twins/tubemq-client-go/client/consumer_impl.go [hunk 5]
```diff
@@ -619,34 +625,43 @@ func (c *consumer) convertMessages(filtered bool, topic string, rsp *protocol.Ge
...
+		checkSum := uint64(crc32.Update(0, crc32.IEEETable, m.GetPayLoadData())) & 0x7FFFFFFF
 		if int32(checkSum) != m.GetCheckSum() {
 			continue
 		}
 		readPos := 0
-		dataLen := len(m.GetPayLoadData())
+		payLoadData := m.GetPayLoadData()
+		dataLen := len(payLoadData)
 		var properties map[string]string
 		if m.GetFlag()&0x01 == 1 {
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/client/consumer_impl.go:636]: Cann't use magic number to here.

**Pull request merged**